src/ax/saga/aggregate.go

```diff
@@ -0,0 +1,147 @@
+package saga
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/jmalloc/ax/src/ax"
+	"github.com/jmalloc/ax/src/internal/visitor"
+)
+
+// Aggregate is a Saga for implementing application-defined domain aggregates.
+type Aggregate struct {
+	ErrorIfNotFound
+	CompletableByData
+
+	Prototype Data
+	Triggers  ax.MessageTypeSet
+	Handle    func(Data, ax.Command, ax.EventRecorder)
+	Apply     func(Data, ax.Event)
+}
+
+// NewAggregate returns a new aggregate saga.
+//
+// Aggregates are a specialization of sagas that handle commands and produce
+// events.
+//
+// It accepts a prototype data instance which is cloned for new instances.
+//
+// For each command type to be handled, the aggregate's data struct must
+// implement a "handler" method that adheres to the following signature:
+//
+//     func (cmd *<T>, rec Recorder)
```

Comment:
Is this the right type of Recorder? Should it be the newly named `EventRecorder`?